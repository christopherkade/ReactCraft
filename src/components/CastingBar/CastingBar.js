import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { BoxDecorators } from "../BoxDecorators"

const Wrapper = styled.div`
  width: ${props => props.width}px;
  height: 1rem;
  position: relative;
  background-color: rgb(47, 27, 37, 0.8);
  box-shadow: inset 1px 1px 7rem 0rem black;
  border: 4px ridge #8C8C8C;
  text-align: center;
`

const Bar = styled.div`
  width: ${props => props.percent}%;
  height: 1rem;
  background-color: #B27A02;
  background: rgb(183,127,0);
  background: linear-gradient(90deg, rgba(183,127,0,1) 96%, rgba(254,255,127,1) 100%);
  box-shadow: inset 1px 1px 0.5rem 0rem black;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
`

const Label = styled.span`
  position: relative;
  margin-left: 2px;
  margin-top: 2px;
  color: white;
  z-index: 2;
  font-size: 1.2rem;
  top: -0.25rem;
`

const CastingBar = ({ width, text, percent }) => {
  return (
    <Wrapper width={width}>
      <Label>
        {text}
      </Label>
      <Bar percent={percent} />
      <BoxDecorators />
    </Wrapper>
  )
}

CastingBar.propTypes = {
  width: PropTypes.number,
  text: PropTypes.string,
  percent: PropTypes.number,
};

CastingBar.defaultProps = {
  width: 150,
  text: "Fireball",
  percent: 0,
};

export default CastingBar
