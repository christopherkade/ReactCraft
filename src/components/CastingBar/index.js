export { default as CastingBar } from "./CastingBar"
