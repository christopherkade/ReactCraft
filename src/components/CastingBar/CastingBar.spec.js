import React from 'react';
import CastingBar from './CastingBar';
import { render } from 'react-testing-library';

describe('<CastingBar />', () => {

  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<CastingBar />)
      expect(container).toMatchSnapshot()
    });
  });

  describe('text prop', () => {
    it('should display Fireball as the default text', () => {
      const { container } = render(<CastingBar />)
      expect(container.textContent).toBe("Fireball")
    })

    it('should display Frostbolt as the text', () => {
      const { container } = render(<CastingBar text="Frostbolt" />)
      expect(container.textContent).toBe("Frostbolt")
    })
  })
});
