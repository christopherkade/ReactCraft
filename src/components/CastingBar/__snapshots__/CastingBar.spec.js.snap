// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CastingBar /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-ifAKCX jewqmR"
    width="150"
  >
    <span
      class="sc-bZQynM ranME"
    >
      Fireball
    </span>
    <div
      class="sc-EHOje bVvRXt"
    />
    <div
      class="sc-bdVaJa fwHBtD"
    />
    <div
      class="sc-bwzfXH dRmJHm"
    />
    <div
      class="sc-htpNat iaYGmT"
    />
    <div
      class="sc-bxivhb fnUviR"
    />
  </div>
</div>
`;
